### PR TITLE
feat: added vm.keyExistsJson

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
## Motivation

I need this to deploy smart contracts. It seems odd that some of the supported VM cheatcodes are missing right now.

### CODE REVIEW

File: crates/evm/traces/src/decoder/mod.rs [hunk 1]
```diff
@@ -447,7 +449,11 @@ impl CallTraceDecoder {
                 } else {
                     let mut decoded = func.abi_decode_input(&data[SELECTOR_LEN..], false).ok()?;
                     let token =
-                        if func.name.as_str() == "parseJson" || func.name.as_str() == "keyExists" {
+
```
Comment (MexicanAce) [inline crates/evm/traces/src/decoder/mod.rs:452]: Lint error: `left behind trailing whitespace`

**Comment by nbaztec** [comment]
@PatrickAlphaC some cheatcodes are missing as we haven't sync'd with upstream foundry yet since March. We're currently in the process of merging upstream. When that happens, the new cheatcodes would automatically be supported.

File: crates/evm/traces/src/decoder/mod.rs (hunk 1 above)
Comment (MexicanAce) [reply crates/evm/traces/src/decoder/mod.rs:452]:
```suggestion

```

**Comment by wottpal** [comment]
This is also bothering us. Would be great to see another upstream merge (plus `--verify` support) soon :)

**Comment by nbaztec** [comment]
Yeah we'll be on it once https://github.com/matter-labs/foundry-zksync/pull/368 gets merged, since it's a pre-requisite for the upstream merge.

**Pull request closed without merging**